onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -307,19 +307,19 @@ Status Pad<float>::Compute(OpKernelContext* ctx) const {
       }
     }
 
-    float value = 0;
+    T value = 0;
     const Tensor* value_tensor = ctx->Input<Tensor>(2);
     if (nullptr != value_tensor) {
-      ORT_ENFORCE(value_tensor->DataType() == DataTypeImpl::GetType<float>() &&
+      ORT_ENFORCE(value_tensor->DataType() == DataTypeImpl::GetType<T>() &&
                       value_tensor->Shape().Size() == 1,
                   "Value tensor should be a 1D tensor of size 1 with the same type as that of the input tensor");
-      value = value_tensor->template Data<float>()[0];
+      value = value_tensor->template Data<T>()[0];
     }
 
-    return PadCpuImpl<float>(ctx, pads, slices, mode_, value);
+    return PadCpuImpl<T, T>(ctx, pads, slices, mode_, value);
   } else {
     // kOnnxDomain Pad opset < 11
-    return PadCpuImpl<float>(ctx, pads_, slices_, mode_, value_);
+    return PadCpuImpl<T, float>(ctx, pads_, slices_, mode_, value_);
```

Comment:
nit: a comment saying why we do this would be great. In opset-2 Pad, the type of value was always `float` irrespective of the `data` input type whereas in opset-11 Pad the type of value is the same type as the `data` input type.

Reply:
Added a comment.